src/resource_manager/gateway.py

```diff
@@ -213,3 +213,16 @@ def gateway_address(self, name: str) -> typing.Optional[str]:
             logger.info("Gateway address not ready, waiting for %s seconds before retrying", delay)
             time.sleep(delay)
         return gateway_address
+
+    def current_gateway_resource(self) -> typing.Optional[GenericNamespacedResource]:
+        """Return the current gateway resource.
+
+        Returns:
+            typing.Optional[GenericNamespacedResource]: The gateway resource
+            or None if the resource does not exists or number of gateway resources != 1.
+        """
+        gateway_resources = self._list_resource()
+        logger.info("%r", gateway_resources)
+        if not gateway_resources or len(gateway_resources) != 1:
```

Comment:
Is it possible to have more than one gateway resource, since there can be only one integration for this charm?

Reply:
Yes it's possible to have more than one gateway resource with the same label selector since we don't have control over the k8s namespace so anyone can create/modify the gateway resource that's managed by the charm. My thought is that in cases like these we can't do much until the gateway resource is updated to its expected state.